packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx

```diff
@@ -46,60 +67,242 @@ export class AtomicSearchBox {
+  }
+
+  private get previousOrLastValue() {
+    if (!this.hasActiveDescendant) {
+      return this.lastValue;
+    }
+
     return (
-      <div class="flex-grow flex items-center">
-        <input
-          placeholder={this.bindings.i18n.t('search')}
-          aria-label={this.bindings.i18n.t('search-box')}
-          type="text"
-          class="h-full outline-none bg-transparent flex-grow px-4 text-neutral-dark placeholder-neutral-dark text-lg"
-          value={this.searchBoxState.value}
-          onFocus={() => (this.hasFocus = true)}
-          onBlur={() => (this.hasFocus = false)}
-          onInput={(e) =>
-            this.searchBox.updateText((e.target as HTMLInputElement).value)
-          }
-        />
-        {this.searchBoxState.isLoading && (
-          <span class="loading w-5 h-5 rounded-full bg-gradient-to-r animate-spin mr-3 grid place-items-center"></span>
-        )}
-        {!this.searchBoxState.isLoading &&
-          this.searchBoxState.value.trim() !== '' && (
-            <Button
-              style="text-transparent"
-              part="clear-button"
-              class="w-8 h-8 mr-1.5 text-neutral-dark"
-              onClick={() => this.searchBox.clear()}
-              ariaLabel={this.bindings.i18n.t('clear')}
-            >
-              <atomic-icon icon={ClearIcon} class="w-3 h-3"></atomic-icon>
-            </Button>
-          )}
-      </div>
+      this.activeDescendantElement?.previousElementSibling || this.lastValue
     );
   }
 
-  // TODO: adapt for atomic-query-suggestions & atomic-recent-queries
-  private renderSuggestions() {
-    if (!this.searchBoxState.suggestions.length) {
+  private focusNextValue() {
+    if (!this.hasSuggestions) {
       return;
     }
 
-    if (!this.hasFocus) {
+    const query = this.nextOrFirstValue.getAttribute('data-value');
+    !isNullOrUndefined(query) && this.updateQuery(query);
+    this.updateActiveDescendant(this.nextOrFirstValue.id);
+  }
+
+  private focusPreviousValue() {
+    if (!this.hasSuggestions) {
+      return;
+    }
+
+    const query = this.previousOrLastValue.getAttribute('data-value');
+    !isNullOrUndefined(query) && this.updateQuery(query);
+    this.updateActiveDescendant(this.previousOrLastValue.id);
+  }
+
+  private onInput(value: string) {
+    this.searchBox.updateText(value);
+    this.updateActiveDescendant();
+  }
+
+  private onFocus() {
+    this.isExpanded = true;
+    if (!this.searchBoxState.suggestions.length) {
+      this.searchBox.showSuggestions();
+    }
+  }
+
+  private onBlur() {
+    this.isExpanded = false;
+    this.updateActiveDescendant();
+  }
+
+  private onSubmit() {
+    if (this.activeDescendantElement) {
+      this.activeDescendantElement.click();
+      this.updateActiveDescendant();
       return;
     }
 
+    this.searchBox.submit();
+    this.updateActiveDescendant();
+  }
+
+  private onKeyup(e: KeyboardEvent) {
```

Comment:
Just wondering why you used `keyUp` to handle submit and clear, and `keyDown` for navigation. Why not the same event for both ?